orchestrator/packages/db/src/api-key-operations/index.ts

```diff
@@ -0,0 +1,20 @@
+import prismaInstance from '../prisma-instance';
+
+const KEY_LENGTH = 64;
+
+export const createAPIKey = async (hostname: string): Promise<string> =>
+  await prismaInstance.$transaction(async (tx) => {
+    const key = generateKey();
+    await tx.apiKey.create({
+      data: {
+        hostname,
+        value: key
+      }
+    });
+    return key;
+  });
+
+export const validAPIKey = async (hostname: string, value: string) =>
+  Boolean(await prismaInstance.apiKey.count({ where: { hostname, value } }))
+
+const generateKey = (): string => [...Array(KEY_LENGTH)].map((_) => Math.floor(Math.random() * 16).toString(16)).join('');
```

Comment:
Since you're on node, use https://nodejs.org/api/crypto.html#crypto_crypto_randombytes_size_callback

Reply:
Done!